cmd/fossa/cmd/test/test_test.go

```diff
@@ -0,0 +1,169 @@
+package test
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/fossas/fossa-cli/api/fossa"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/cli"
+)
+
+const orgID = 3
+
+// taskStatus is a struct that imitates the anonymous struct within fossa.Build
+type taskStatus struct {
+	Status string
+}
+
+func TestRunCustomFetcherCustomProject(t *testing.T) {
+	c := make(chan string)
+	locator := "custom+" + strconv.Itoa(orgID) + "%2FtestRun$1000"
+	ts := testServer(locator, c)
+
+	flagSet := testFlags("custom", "testRun", ts.URL, 1000)
+	context := cli.NewContext(&cli.App{}, flagSet, &cli.Context{})
+
+	go Run(context)
+	msg := <-c
+	if msg != "SUCCESS" {
+		assert.Equal(t, locator, msg)
```

Comment:
Why `assert.Equal` instead of `t.Fail`?

Reply:
This is one area I had the most difficulty making clear. The output from `t.Fail` running `go test -v` is 
```
=== RUN   TestRunCustomFetcherCustomProject
⣾ Waiting for analysis to complete... --- FAIL: TestRunCustomFetcherCustomProject (0.00s)
```

Using a goroutine and `assert.Equal` I can improve the feedback to something that is much more descriptive of the actual error.
```
=== RUN   TestRunCustomFetcherCustomProject
⣾ Waiting for analysis to complete... --- FAIL: TestRunCustomFetcherCustomProject (0.00s)
    test_test.go:35:
        	Error Trace:	test_test.go:35
        	Error:      	Not equal:
        	            	expected: "ctom+3%2FtestRun$1000"
        	            	actual  : "/api/cli/custom+3%2FtestRun$1000/latest_build"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-ctom+3%2FtestRun$1000
        	            	+/api/cli/custom+3%2FtestRun$1000/latest_build
        	Test:       	TestRunCustomFetcherCustomProject
```

Reply:
I think there are two good options here:

- Use `log.Debug` (or `t.Log`) to describe the locator. This output will show up on `go test -v`.
- Construct the actual expected locator, and then compare against that in `assert.Equal`. That way your `assert.Equal` is actually semantically an equality assertion.

My personal preference is for the latter. I think that provides good feedback while remaining clear.

Reply:
I agree with this, I've been thinking of better ways to clarify this. The problem with comparing only the locators is that if the test fails it assumes the rest of the path is correct which might not be the case. I'm comfortable going down the route of asserting only against the locator in this case however because that is what this test is varying between cases.